### ISSUE TITLE
Result of symbolic products grows uncontrollably

### ISSUE THREAD

**Issue by jdbancal** [replab/replab#247]
The test `replab/quantum/QuantumTest.m` doesn't finish after more than 2 hours of computation (on matlab). More precisely, it is stuck in the following section:
```
    [G rep] = replab.quantum.clifford_qudit(3);
    test_suite = replab.FiniteGroupLaws(G).addTestCases(test_suite);
    test_suite = replab.RepLaws(rep).addTestCases(test_suite);
```
The reason seems to be that clifford groups are defined in `+replab/+quantum/clifford_qubit.m` in terms of symbolic objects, and composition of symbolic objects degenerates. Concretely, after some time, call to `+replab.UnitaryGroup.compose` are made with arguments `X` and `Y` such as
```
X =
 
[                                                                                                                                                         ((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3)*(- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*1i)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^3)/3) - ((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^2)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3) - ((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*2i)/3)*((2*exp((pi*2i)/9))/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2))/3),                                                                                                                                                       - ((2*exp((pi*2i)/9))/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3) + (- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*1i)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^3)/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^2)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3),                                                                                                                                                       - ((2*exp((pi*2i)/9))/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3) + ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3)*(- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*1i)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^3)/3) - ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 + 1/2)^2)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3)]
[ - ((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3)*(exp((pi*2i)/9)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^3)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2))/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)*1i)/3)*((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3) - ((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*2i)/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*1i)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2))/3), - ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3)*(exp((pi*2i)/9)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^3)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2))/3) - ((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*1i)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2))/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)*1i)/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3), - ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3)*(exp((pi*2i)/9)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^3)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2))/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*1i)/3 - (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)*1i)/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3)]
[ ((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3)*(- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^3*1i)/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3) - ((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*2i)/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3), ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3)*(- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^3*1i)/3) - ((3^(1/2)*exp((pi*7i)/9)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3)*(exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3), ((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3))/3)*(- exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*1i)/3 + (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)^3*1i)/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2)*1i)/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6))/3) - (exp((pi*2i)/9)/3 + (exp((pi*2i)/9)*((3^(1/2)*1i)/2 - 1/2)^2*((3^(1/2)*1i)/2 + 1/2)^2)/3 - (exp((pi*2i)/9)*(3^(1/2)/2 - 1i/2)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 + 1/2))/3)*((3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)^3*(((3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)^2*2i)/3 + 1/3)*1i)/3 + (3^(1/2)*exp((pi*7i)/9)*((3^(1/2)*1i)/2 - 1/2)*((3^(1/2)*1i)/6 + ((3^(1/2)*1i)/2 - 1/2)^2/3 + 1/6)*1i)/3 - (3^(1/2)*exp((pi*7i)/9)*(3^(1/2)/2 + 1i/2)*((3^(1/2)*1i)/2 - 1/2)*(1/3 + ((3^(1/2)*1i)/2 - 1/2)^4/3 - (3^(1/2)/2 + 1i/2)^2/3))/3)]
```
whereas the object at hand actually is quite simple:
```
>> simplify(X)
 
ans =
 
[ (3^(1/2)*1i)/3,   1/2 - (3^(1/2)*1i)/6, - (3^(1/2)*1i)/6 - 1/2]
[ (3^(1/2)*1i)/3, - (3^(1/2)*1i)/6 - 1/2,   1/2 - (3^(1/2)*1i)/6]
[ (3^(1/2)*1i)/3,         (3^(1/2)*1i)/3,         (3^(1/2)*1i)/3]
```
As a result, computations become extra-ordinarily slow, and the above section of the tests doesn't seem to finish. How should we deal with that?

One possibility could be to modify `+replab.UnitaryGroup.compose` as follows:
```
        function Z = compose(self, X, Y)
            Z = X * Y;
            if isa(Z, 'sym')
                Z = simplify(Z);
            end
        end
```
However, this would have a small but systematic overhead on all other compositions. Also, it is not clear whether symbolic expressions would still be growing in other places in the code...

**Comment by jdbancal** [comment]
As a temporary fix, we could also simply disable this part of the test, and restore it once we have more experience with sym integration and we have a clear strategy for how we should incorporate the symbolic toolbox?

**Comment by denisrosset** [comment]
I've tried to add a call to `simplify`... and it is still really slow. While the Symbolic toolbox works OK with rational numbers, it is catastrophically slow over the cyclotomic field (which is what we are using).

I think we can remove temporarily the Clifford group functionality: it is no use having something either 1) with bad numerical behaviour (as before) or 2) which requires a minute to an hour to construct.

Future fixes could include:

1. A smarter construction of the BSGS chain with its images. Right now we use a randomized Schreier-Sims algorithm that computes a lot of products to increase randomness. To construct the BSGS chain of a group of order 192, we perform around 4000 products, which seems unnecessary. That code can be heavily optimized.

2. An interface from Matlab to an optimized exact arithmetic library over the cyclotomics, as they are prevalent in finite group representations.

**Comment by jdbancal** [comment]
All right! I agree a fully satisfying solution would involve significant optimization work. So for now I simply disabled the problematic test (code for the Clifford groups is still present).

As it turns out, other tests in `replab/sym/SymmetricGroupIrrepsTest.m` were also taking hours to run on matlab (all other RepLAB tests combined are done in less than 3 mn). So I also commented these, citing this issue as a reason for reference.
